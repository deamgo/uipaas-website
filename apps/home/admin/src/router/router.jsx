import { createBrowserRouter, Navigate } from 'react-router-dom'
import App from '../App.jsx'
import Signin from '../layout/Signin'

export const router = createBrowserRouter([
    {
        path: '/',
        Component: App,
        errorElement: <Navigate to="/" />,
    },
    {
        path: '/logo',
        Component: Signin,
    },
])