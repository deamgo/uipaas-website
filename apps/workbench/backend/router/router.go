package router

import (
	"net/http"

	"github.com/deamgo/workbench/api/account"
	"github.com/deamgo/workbench/api/application"
	"github.com/deamgo/workbench/api/devdepot"
	"github.com/deamgo/workbench/api/developer"
	"github.com/deamgo/workbench/api/workspace"
	"github.com/deamgo/workbench/api/ws"
	"github.com/deamgo/workbench/context"
	"github.com/deamgo/workbench/middleware"

	"github.com/gin-gonic/gin"
)

func NewRouter(ctx context.ApplicationContext) http.Handler {
	e := gin.New()
	e.Use(gin.Recovery())
	e.Use(middleware.Cors())
	e.GET("/v1/ws", ws.WSHandler())
	mountAPIs(e, ctx)
	return e
}
func mountAPIs(e *gin.Engine, ctx context.ApplicationContext) {
	api := e.Group("v1")
	{
		api.POST("/signup", account.SignUp(ctx))
		api.POST("/signup_verify", account.SignUpVerify(ctx))
		api.POST("/signin", account.SignIn(ctx))
		api.POST("/forgot_verify", account.SendForgotVerify(ctx))
		api.PUT("/reset_password", account.ResetPassword(ctx))
		api.GET("/logout", middleware.JWTAuthMiddleware(), account.Logout())
	}
	developerAPI := e.Group("v1", middleware.JWTAuthMiddleware())
	{

		// Get developer
		developerAPI.GET("/developer", developer.DeveloperGetByID(ctx))
		// Modify the username
		developerAPI.PUT("/developer/username/:id", developer.DeveloperNameModify(ctx))
		// Modify the email
		developerAPI.POST("/developer/email/firststep", developer.VerifyEmailAndPwd(ctx))
		developerAPI.POST("/developer/email/secondstep", developer.SendModifyEmailVerify(ctx))
		developerAPI.PUT("/developer/email/thirdstep", developer.VerifyEmailVerificationCode(ctx))
		// Modify the password
		developerAPI.POST("/developer/password/firststep", developer.SendModifyPwdVerify(ctx))
		developerAPI.POST("/developer/password/secondstep", developer.VerifyPwdVerificationCode(ctx))
		developerAPI.PUT("/developer/password/thirdstep", developer.ModifyPwd(ctx))
	}
	workspaceApi := api.Group("/workspace", middleware.JWTAuthMiddleware())
	{
		workspaceApi.DELETE("/:workspace_id/settings", middleware.AuthMiddleware(), workspace.WorkspaceDel(ctx))
		workspaceApi.PUT("/:workspace_id/settings", middleware.AuthMiddleware(), workspace.WorkspaceNameModify(ctx))
		workspaceApi.POST("/create", workspace.WorkspaceCreate(ctx))
		workspaceApi.GET("/list", workspace.WorkspaceGetListById(ctx))
		workspaceApi.POST("/logo", workspace.WorkspaceGetLogoPath(ctx))
		workspaceApi.GET("/:workspace_id/developer", middleware.AuthMiddleware(), devdepot.DevdepotList(ctx))
		workspaceApi.GET("/:workspace_id/developer/search", middleware.AuthMiddleware(), devdepot.DevdepotSearch(ctx))
		workspaceApi.DELETE("/:workspace_id/developer", middleware.AuthMiddleware(), devdepot.DevdepotDel(ctx))
		workspaceApi.PUT("/:workspace_id/developer", middleware.AuthMiddleware(), devdepot.DevdepotRoleModify(ctx))
		workspaceApi.POST("/:workspace_id/developer/invite", middleware.AuthMiddleware(), devdepot.DevDepotInvite(ctx))

	}
	applicationAPI := e.Group("v1", middleware.JWTAuthMiddleware(), middleware.AuthMiddleware())
	{
		applicationAPI.POST("/workspace/:workspace_id/application", application.ApplicationCreate(ctx))
		applicationAPI.GET("/workspace/:workspace_id/application", application.ApplicationList(ctx))
		applicationAPI.GET("/workspace/:workspace_id/application/search", application.ApplicationSearch(ctx))
	}

}
