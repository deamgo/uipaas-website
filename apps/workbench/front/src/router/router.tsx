import { RouteObject, createBrowserRouter, redirect } from 'react-router-dom'
//layout
import Layout from '@/views/layout'
import Sign from '@views/sign'
import SignUp from '@/views/sign/SignUp'
import EmailVerif from '@/views/sign/EmailVerif'
import SignIn from '@/views/sign/SignIn'
import PwdReset from '@/views/sign/PwdReset'
import Privacy from '@/views/privacy'
import ContentApp from '@/views/layout/content-app'
import ContentUsr from '@/views/layout/content-usr'
import UserProfile from '@/views/layout/content-usr/content/profile'
import ContentWorkSpace from '@/views/layout/content-ws-sys'
import WSDevelopers from '@/views/layout/content-ws-sys/content/developers'
import WSSettings from '@/views/layout/content-ws-sys/content/settings'

const tokenLoader = async () => {
  const token = sessionStorage.getItem('token')
  if (!token) {
    return redirect('/s')
  }
  return null
}

export const routes: RouteObject[] = [
  {
    path: '/',
    Component: Layout,
    loader: tokenLoader,
    children: [
      {
        index: true,
        Component: ContentApp,
      },
      {
        path: '/u',
        Component: ContentUsr,
        children: [
          {
            index: true,
            Component: UserProfile,
          }
        ]
      },
      {
        path: '/workspace',
        Component: ContentWorkSpace,
        children: [
          {
            index: true,
            Component: WSDevelopers,
          },
          {
            path: '/workspace/settings',
            Component: WSSettings,
          }
        ]
      },
    ]
  },
  {
    path: '/s',
    Component: Sign,
    children: [
      {
        path: '/s/up',
        Component: SignUp,
      },
      {
        path: '/s/ev',
        Component: EmailVerif
      },
      {
        index: true,
        Component: SignIn
      },
      {
        path: '/s/ryp',
        Component: PwdReset
      }
    ]
  },
  {
    path: '/privacy',
    Component: Privacy
  }
]


export const router = createBrowserRouter(routes)
