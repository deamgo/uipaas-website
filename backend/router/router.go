package router

import (
	"github.com/deamgo/uipass-waitlist-page/backend/api/companyinfo"
	"net/http"

	"github.com/deamgo/uipass-waitlist-page/backend/api/user"
	"github.com/deamgo/uipass-waitlist-page/backend/context"

	"github.com/gin-gonic/gin"
)

func NewRouter(ctx context.ApplicationContext) http.Handler {
	e := gin.New()
	// add middleware
	e.Use(gin.Recovery())
	mountAPIs(e, ctx)

	return e
}
func mountAPIs(e *gin.Engine, ctx context.ApplicationContext) {
	api := e.Group("api")
	{
		api.GET("/user/:id", user.UserGet(ctx))
		api.POST("/login", user.UserLogin(ctx))
		api.GET("/companyinfo", companyinfo.CompanyInfoGet(ctx))
	}

}
