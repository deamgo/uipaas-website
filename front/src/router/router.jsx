import { createBrowserRouter, Navigate } from 'react-router-dom'
import App from '../App.jsx'
import PriPolicy from '../layout/privacyPolicy.jsx'
// import ErrorPage from '../layout/errorPage.jsx'

export const router = createBrowserRouter([
    {
        path: '/',
        Component: App,
        errorElement: <Navigate to="/" />,
    },
    {
        path: '/privacy',
        Component: PriPolicy,
    },
])