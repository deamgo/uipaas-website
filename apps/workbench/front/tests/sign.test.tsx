import { RouterProvider, createMemoryRouter } from 'react-router-dom'

import { routes } from '@/router/router'
import { render } from '@testing-library/react'

it('Sign page Test', () => {
  const router = createMemoryRouter(
    routes,
    {
      initialEntries: ['/s/up'],
      initialIndex: 1,
    }
  )
  const SignIN = render(<RouterProvider router={router} />)
  expect(SignIN).toMatchSnapshot()
})