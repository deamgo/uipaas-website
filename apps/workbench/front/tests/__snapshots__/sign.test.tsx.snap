// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sign page Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      class="message-wrap"
      id="message-wrap"
    />
    <div>
      <div
        class="__sign"
      >
        <div
          class="__sign_logo"
        >
          <img
            alt="UIPaaS"
            src=""
          />
        </div>
        <div
          class="__sign_container"
        >
          <div
            class="__sign_title"
          >
            <span>
              Sign up
            </span>
            <span>
              Complete the Information for Account Registration
            </span>
          </div>
          <div
            class="__sign_form"
          >
            <div
              class="__sign_form_input"
            >
              <div
                class="__input_wrapper"
              >
                <div
                  class="__input_wrapper_title"
                >
                  Invitation code
                  <span>
                    *
                  </span>
                </div>
                <div
                  class="__input_wrapper_main"
                >
                  <input
                    id="1"
                    name=""
                    placeholder="Enter your invitation code"
                    type="text"
                  />
                </div>
                <div
                  class="__input_wrapper_invalid"
                >
                  Please enter the invitation code
                </div>
              </div>
            </div>
            <div
              class="__sign_form_input"
            >
              <div
                class="__input_wrapper"
              >
                <div
                  class="__input_wrapper_title"
                >
                  Name
                  <span>
                    *
                  </span>
                </div>
                <div
                  class="__input_wrapper_main"
                >
                  <input
                    id="2"
                    name=""
                    placeholder="Enter your name"
                    type="text"
                  />
                </div>
                <div
                  class="__input_wrapper_invalid"
                >
                  Please enter your name
                </div>
              </div>
            </div>
            <div
              class="__sign_form_input"
            >
              <div
                class="__input_wrapper"
              >
                <div
                  class="__input_wrapper_title"
                >
                  Email
                  <span>
                    *
                  </span>
                </div>
                <div
                  class="__input_wrapper_main"
                >
                  <input
                    id="3"
                    name=""
                    placeholder="Enter your email address"
                    type="text"
                  />
                </div>
                <div
                  class="__input_wrapper_invalid"
                >
                  Please enter your email address
                </div>
              </div>
            </div>
            <div
              class="__sign_form_input"
            >
              <div
                class="__input_wrapper"
              >
                <div
                  class="__input_wrapper_title"
                >
                  Password
                  <span>
                    *
                  </span>
                </div>
                <div
                  class="__input_wrapper_main"
                >
                  <input
                    id="4"
                    name=""
                    placeholder="Enter new password"
                    type="password"
                  />
                  <div
                    class="__input_wrapper_main_show"
                  >
                    <img
                      src=""
                    />
                    <div
                      class="__show_closed"
                    />
                  </div>
                </div>
                <div
                  class="__input_wrapper_invalid"
                >
                  Please enter your password
                </div>
              </div>
            </div>
          </div>
          <div
            class="__sign_continue"
          >
            <div
              class="__sign_continue_tip"
            >
              <span>
                Already have an account?
              </span>
              <span>
                <a
                  href="/s/in"
                >
                  Sign in
                </a>
              </span>
            </div>
            <button
              class="__button_wrapper"
              disabled=""
              type="button"
            >
              Continue
            </button>
            <div
              class="__sign_continue_privacy"
            >
              <span>
                By using UIPaaS, you are agreeing to the
              </span>
              <span>
                <a
                  href="/privacy"
                  target="_blank"
                >
                  Privacy Policy.
                </a>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="__sign"
    >
      <div
        class="__sign_logo"
      >
        <img
          alt="UIPaaS"
          src=""
        />
      </div>
      <div
        class="__sign_container"
      >
        <div
          class="__sign_title"
        >
          <span>
            Sign up
          </span>
          <span>
            Complete the Information for Account Registration
          </span>
        </div>
        <div
          class="__sign_form"
        >
          <div
            class="__sign_form_input"
          >
            <div
              class="__input_wrapper"
            >
              <div
                class="__input_wrapper_title"
              >
                Invitation code
                <span>
                  *
                </span>
              </div>
              <div
                class="__input_wrapper_main"
              >
                <input
                  id="1"
                  name=""
                  placeholder="Enter your invitation code"
                  type="text"
                />
              </div>
              <div
                class="__input_wrapper_invalid"
              >
                Please enter the invitation code
              </div>
            </div>
          </div>
          <div
            class="__sign_form_input"
          >
            <div
              class="__input_wrapper"
            >
              <div
                class="__input_wrapper_title"
              >
                Name
                <span>
                  *
                </span>
              </div>
              <div
                class="__input_wrapper_main"
              >
                <input
                  id="2"
                  name=""
                  placeholder="Enter your name"
                  type="text"
                />
              </div>
              <div
                class="__input_wrapper_invalid"
              >
                Please enter your name
              </div>
            </div>
          </div>
          <div
            class="__sign_form_input"
          >
            <div
              class="__input_wrapper"
            >
              <div
                class="__input_wrapper_title"
              >
                Email
                <span>
                  *
                </span>
              </div>
              <div
                class="__input_wrapper_main"
              >
                <input
                  id="3"
                  name=""
                  placeholder="Enter your email address"
                  type="text"
                />
              </div>
              <div
                class="__input_wrapper_invalid"
              >
                Please enter your email address
              </div>
            </div>
          </div>
          <div
            class="__sign_form_input"
          >
            <div
              class="__input_wrapper"
            >
              <div
                class="__input_wrapper_title"
              >
                Password
                <span>
                  *
                </span>
              </div>
              <div
                class="__input_wrapper_main"
              >
                <input
                  id="4"
                  name=""
                  placeholder="Enter new password"
                  type="password"
                />
                <div
                  class="__input_wrapper_main_show"
                >
                  <img
                    src=""
                  />
                  <div
                    class="__show_closed"
                  />
                </div>
              </div>
              <div
                class="__input_wrapper_invalid"
              >
                Please enter your password
              </div>
            </div>
          </div>
        </div>
        <div
          class="__sign_continue"
        >
          <div
            class="__sign_continue_tip"
          >
            <span>
              Already have an account?
            </span>
            <span>
              <a
                href="/s/in"
              >
                Sign in
              </a>
            </span>
          </div>
          <button
            class="__button_wrapper"
            disabled=""
            type="button"
          >
            Continue
          </button>
          <div
            class="__sign_continue_privacy"
          >
            <span>
              By using UIPaaS, you are agreeing to the
            </span>
            <span>
              <a
                href="/privacy"
                target="_blank"
              >
                Privacy Policy.
              </a>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
